test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java

```diff
@@ -259,17 +272,43 @@ protected Transaction finalizedTxn(
 			txn = getSigned(spec, spec.txns().getReadyToSign(netDef), keys);
 		}
 
-		return txn;
+		return finalizedTxnFromTxnWithBodyBytesAndSigMap(txn);
 	}
 
-	private Transaction getSigned(HapiApiSpec spec, Transaction.Builder builder, List<Key> keys) throws Throwable {
-		if (useLegacySignature) {
-			return spec.keys().listSign(builder, keys, overrides);
-		} else {
-			return sigMapGen.isPresent()
-					? spec.keys().sign(builder, keys, overrides, sigMapGen.get())
-					: spec.keys().sign(builder, keys, overrides);
+	private Transaction finalizedTxnFromTxnWithBodyBytesAndSigMap(Transaction txnWithBodyBytesAndSigMap
+	) throws Throwable {
+		if (asTxnWithOnlySigMap) {
+			return txnWithBodyBytesAndSigMap.toBuilder().clearBodyBytes().build();
+		}
+
+		ByteString bodyByteString = CommonUtils.extractTransactionBodyByteString(txnWithBodyBytesAndSigMap);
+		SignatureMap sigMap = CommonUtils.extractSignatureMap(txnWithBodyBytesAndSigMap);
+		SignedTransaction signedTransaction = SignedTransaction.newBuilder()
+				.setBodyBytes(bodyByteString)
+				.setSigMap(sigMap)
+				.build();
+		Transaction.Builder txnWithSignedTxnBytesBuilder =
+				Transaction.newBuilder().setSignedTransactionBytes(signedTransaction.toByteString());
+
+		if (asTxnWithSignedTxnBytesAndSigMap) {
+			return txnWithSignedTxnBytesBuilder.setSigMap(sigMap).build();
 		}
+
+		if (asTxnWithSignedTxnBytesAndBodyBytes) {
+			return txnWithSignedTxnBytesBuilder.setBodyBytes(bodyByteString).build();
+		}
```

Comment:
I am not use I understand the purposes of these two if statements here.  Since 'signedTransaction' is already constructed as having bodyByteString and sigMap. Now we construct a Transaction instance with 'signedTranaction' as its member, and meanwhile, set bodyBytesString and/or sigMap to the new Transaction instance. Do we have such use cases in real world?

Reply:
Please read this PR description and the issue #432, when this happens, the node should return `INVALID_TRANSACTION`

Reply:
Yeah, I was thinking this might be used to construct some negative test scenarios. Thx for the clarification.